src/dhcp_mon.cpp

```diff
@@ -153,49 +151,76 @@ static void timeout_callback(evutil_socket_t fd, short event, void *arg)
 }
 
 /**
- * @code dhcp_mon_init(window_sec, max_count);
+ * @code update_counter(dhcp_packet_direction_t dir)
+ * @brief Function to update counter in COUNTERS_DB
+ * @param dir       Packet direction
+ * @return none
+ */
+void update_counter(dhcp_packet_direction_t dir) {
+    std::unordered_map<std::string, std::unordered_map<uint8_t, uint64_t>>* counter = dhcp_device_get_counter(dir);
+    std::string table_name;
+    for (const auto& outer_pair : *counter) {
+        const std::string interface_name = outer_pair.first;
+        const auto* inner_map = &outer_pair.second;
+        std::string value = generate_json_string(inner_map);
+        /**
+         * Only add downstream prefix for non-downstream interface
+         */
+        if (downstream_if_name.compare(interface_name) != 0) {
+            table_name = DB_COUNTER_TABLE + downstream_if_name + "_" + interface_name;
+        } else {
+            table_name = DB_COUNTER_TABLE + interface_name;
+        }
+        mCountersDbPtr->hset(table_name, (dir == DHCP_RX) ? "RX" : "TX", value);
+    }
+}
+
+/**
+ * @code db_update_callback(fd, event, arg);
+ *
+ * @brief periodic timer call back
+ *
+ * @param fd        libevent socket
+ * @param event     event triggered
+ * @param arg       pointer user provided context (libevent base)
+ *
+ * @return none
+ */
+static void db_update_callback(evutil_socket_t fd, short event, void *arg)
+{
+    update_counter(DHCP_RX);
+    update_counter(DHCP_TX);
+}
+
+/**
+ * @code dhcp_mon_init(window_sec, max_count, dbu_update_interval);
  *
  * initializes event base and periodic timer event that continuously collects dhcp relay health status every window_sec
  * seconds. It also writes to syslog when dhcp relay has been unhealthy for consecutive max_count checks.
  *
  */
-int dhcp_mon_init(int window_sec, int max_count)
+int dhcp_mon_init(int window_sec, int max_count, int db_update_interval)
 {
     int rv = -1;
 
     do {
         window_interval_sec = window_sec;
         dhcp_unhealthy_max_count = max_count;
+        db_update_interval_sec = db_update_interval;
 
-        base = event_base_new();
-        if (base == NULL) {
-            syslog(LOG_ERR, "Could not initialize libevent!\n");
-            break;
-        }
+        evthread_use_pthreads();
 
-        ev_sigint = evsignal_new(base, SIGINT, signal_callback, base);
-        if (ev_sigint == NULL) {
-            syslog(LOG_ERR, "Could not create SIGINT libevent signal!\n");
+        main_event_mgr = new event_mgr("MAIN");
```

Comment:
Do we have protection to avoid re-enter `dhcp_mon_init`? 
What if the `main_event_mgr` already have an object assigned?

Reply:
Added break